### PR TITLE
Be 14 add csv export

### PR DESCRIPTION
**Pull request by Randayy**
civ export 
image
getting all user result in one company 
image
getting some user results in some company 
image

### CODE REVIEW

File: app/services/quiz_service.py [hunk 1]
```diff
@@ -124,6 +131,69 @@ async def check_if_user_alredy_submitted_quiz(self, current_user_id: UUID, quiz_
...
+        async for key in redis_client.scan_iter(f'user_answer:{user_id}:{quiz_id}:*'):
+            data = await redis_client.get_data(key)
+            data = json.loads(data)
+            user_answer_records.append(data)
+
+        await self.save_user_answers_to_csv(user_answer_records)
+        return user_answer_records
+
+    async def save_user_answers_to_csv(self, user_answer_records: list):
+        with open('user_answer_records.csv', 'w', newline='') as csvfile:
```
Comment (Artemooon) [inline app/services/quiz_service.py:162]: where are the header columns for the CSV?

File: app/services/quiz_service.py [hunk 2]
```diff
@@ -124,6 +131,69 @@ async def check_if_user_alredy_submitted_quiz(self, current_user_id: UUID, quiz_
...
+            fieldnames = user_answer_records[0].keys()
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+
+            writer.writeheader()
+            for record in user_answer_records:
+                writer.writerow(record)
```
Comment (Artemooon) [inline app/services/quiz_service.py:168]: where is the JSON file export?

Comment (Randayy) [reply app/services/quiz_service.py:168]: there should be file json created? I thought I should be returned as json , ok I will fix

File: app/services/quiz_service.py (hunk 1 above)
Comment (Randayy) [reply app/services/quiz_service.py:162]: file looks like such 
image
headers should be names of rows ?

File: app/services/quiz_service.py (hunk 2 above)
Comment (Randayy) [reply app/services/quiz_service.py:168]: added

**Comment by Randayy** [comment]
image

File: app/services/quiz_service.py [hunk 3]
```diff
@@ -124,6 +131,74 @@ async def check_if_user_alredy_submitted_quiz(self, current_user_id: UUID, quiz_
...
+
+    async def get_all_user_answer_records(self, user_id: UUID, quiz_id: UUID):
+        redis_client = RedisClient()
+        user_answer_records = []
+        async for key in redis_client.scan_iter(f'user_answer:{user_id}:{quiz_id}:*'):
+            data = await redis_client.get_data(key)
+            data = json.loads(data)
+            user_answer_records.append(data)
+
+        await self.save_user_answers_to_csv(user_answer_records)
```
Comment (Artemooon) [inline app/services/quiz_service.py]: add a flag in the router which export should be done, csv or json
create enum ExportType and export only that type which was selected by user.

Comment (Randayy) [reply app/services/quiz_service.py]: fixed

**Review by Artemooon — APPROVED** [review]

**Pull request merged**